assets/queries/k8s/anonymous_auth_is_not_set_to_false/query.rego

```diff
@@ -0,0 +1,45 @@
+package Cx
+
+import data.generic.k8s as k8sLib
+
+CxPolicy[result] {
+	resource := input.document[i]
+	specInfo := k8sLib.getSpecInfo(resource)
+	container := specInfo.spec.containers[j]
+	commands := ["kube-apiserver", "kubelet"]
+
+	inArray(container.command, commands[_])
+	containerArgs := object.get(container, "args", {})
+	inArray(containerArgs, "--anonymous-auth=true")
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("spec.container[%d].args", [j]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "--anonymous-auth flag should be set to false",
+		"keyActualValue": "--anonymous-auth flag is set to true",
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i]
+	specInfo := k8sLib.getSpecInfo(resource)
+	container := specInfo.spec.containers[j]
+	commands := ["kube-apiserver", "kubelet"]
+
+	inArray(container.command, commands[_])
+	containerArgs := object.get(container, "args", {})
```

Comment:
Please, use the function `get_nested_values_info` in line 31. It receives two inputs: object and array_vals. This function verifies if the array_vals are nested in the object. It will return an object with two fields: valid (returns if the array_vals are nested in the object (array_vals should be sorted) and searchKey returns the searchKey possible.

For example:

```
object := {"elem1": {"elem2": "elem3"}}
array_vals := ["elem2", "elem3", "elem4"]
return_value := {"valid": false, "searchKey": "elem2.elem3"}
```
You can find an example of the application [here](https://github.com/Checkmarx/kics/blob/master/assets/queries/k8s/containers_run_with_low_uid/query.rego#L42-L50)